cli/pack/systemd_dir.go

```diff
@@ -217,25 +228,122 @@ func getSystemdTemplate(ctx *context.Ctx) (templates.Template, error) {
+		mapUnitEnv[tarantoolEnvKey] = value
+	}
+
+	return nil
+}
+
+func setTarantoolEnvValues(ctx *context.Ctx, systemdCtx *map[string]interface{}, systemdUnitParams *SystemdUnitParams) error {
+	setDefaultTarantoolEnvValues(ctx, systemdCtx)
+
+	if err := updateUnitEnvBySpecifiedArgs((*systemdCtx)["UnitEnv"], (*systemdUnitParams).InstanceEnv); err != nil {
+		return err
+	}
+
+	if err := updateUnitEnvBySpecifiedArgs((*systemdCtx)["InstUnitEnv"], (*systemdUnitParams).InstanceEnv); err != nil {
+		return err
+	}
+
+	if err := updateUnitEnvBySpecifiedArgs((*systemdCtx)["StateboardUnitEnv"], (*systemdUnitParams).StateboardEnv); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getSpecifiedStringArg(defaultValue string, unitEnv UnitEnvArgs, argName string) (string, error){
+	result := defaultValue
+
+	if value, ok := unitEnv[argName]; ok {
+		arg, ok := value.(string)
+		if !ok {
+			return "", fmt.Errorf("%s parameter type should be string", argName)
+		}
+		result = arg
+	}
+
+	return result, nil
+}
+
+func getSystemdCtx(ctx *context.Ctx, systemdUnitParams *SystemdUnitParams) (*map[string]interface{}, error) {
+	var err error
+
 	systemdCtx := make(map[string]interface{})
 
-	systemdCtx["Name"] = ctx.Project.Name
-	systemdCtx["StateboardName"] = ctx.Project.StateboardName
+	systemdCtx["Name"], err = getSpecifiedStringArg(ctx.Project.Name, (*systemdUnitParams).InstanceEnv, "app-name")
+	if err != nil {
+		return nil, err
+	}
 
-	systemdCtx["DefaultWorkDir"] = project.GetInstanceWorkDir(ctx, "default")
-	systemdCtx["InstanceWorkDir"] = project.GetInstanceWorkDir(ctx, instanceNameSpecifier)
-	systemdCtx["StateboardWorkDir"] = project.GetStateboardWorkDir(ctx)
+	systemdCtx["StateboardName"], err = getSpecifiedStringArg(ctx.Project.StateboardName, (*systemdUnitParams).StateboardEnv, "app-name")
+	if err != nil {
+		return nil, err
+	}
 
-	systemdCtx["DefaultPidFile"] = project.GetInstancePidFile(ctx, "default")
-	systemdCtx["InstancePidFile"] = project.GetInstancePidFile(ctx, instanceNameSpecifier)
-	systemdCtx["StateboardPidFile"] = project.GetStateboardPidFile(ctx)
+	systemdCtx["DefaultWorkDir"], err = getSpecifiedStringArg(project.GetInstanceWorkDir(ctx, "default"), (*systemdUnitParams).InstanceEnv, "workdir")
+	if err != nil {
+		return nil, err
+	}
 
-	systemdCtx["DefaultConsoleSock"] = project.GetInstanceConsoleSock(ctx, "default")
-	systemdCtx["InstanceConsoleSock"] = project.GetInstanceConsoleSock(ctx, instanceNameSpecifier)
-	systemdCtx["StateboardConsoleSock"] = project.GetStateboardConsoleSock(ctx)
+	systemdCtx["InstanceWorkDir"], err = getSpecifiedStringArg(project.GetInstanceWorkDir(ctx, instanceNameSpecifier), (*systemdUnitParams).InstanceEnv, "workdir")
+	if err != nil {
+		return nil, err
+	}
 
-	systemdCtx["ConfPath"] = ctx.Running.ConfPath
+	systemdCtx["StateboardWorkDir"], err = getSpecifiedStringArg(project.GetStateboardWorkDir(ctx), (*systemdUnitParams).StateboardEnv, "workdir")
```

Comment:
What about reducing copy-paste?

This snippet has more lines, but it's easier to read and understand:

```go
systemdCtxParams := []systemdCtxParam{
    {
        ArgName: "app-name",
        CtxKey: "Name",
        DefaultValue: ctx.Project.Name,
        EnvArgs: (*systemdUnitParams).InstanceEnv,
    },
   ...
}

for _, systemdCtxParam := range systemdCtxParams {
    systemdCtx[systemdCtxParam.CtxKey], err = getSpecifiedStringArg(...)
	if err != nil {
		return nil, err
	}
}
```